### ISSUE TITLE
Unable to set { additionalProperties: true } on Type.Object().

### ISSUE THREAD

**Issue by dsmyda** [sinclairzx81/typebox#73]
Hello! I recently bumped my version of typebox from 0.12.0 to latest, and since then I've lost the ability to set additionalProperties to true on an object type. 

I've narrowed down the change in behavior from 0.14.1 -> 0.15.0. 
Suspected commit: https://github.com/sinclairzx81/typebox/commit/2657b0235cbae1dd4c25e129db9faf5b6594b9fa

Here's is a minimal reproducible example:

const response = Type.Object({ 
	payload: Type.Object({}, {additionalProperties: true})
})

console.log(JSON.stringify(response, null, 2))

Version 0.14.1
`    {
      "type": "object",
      "properties": {
        "payload": {
          "additionalProperties": true,
          "type": "object",
          "properties": {}
        }
      },
      "required": [
        "payload"
      ]
    }`

Version 0.15.0
`      {
        "type": "object",
        "additionalProperties": false,
        "properties": {
          "payload": {
            "additionalProperties": false,
            "type": "object",
            "properties": {}
          }
        },
        "required": [
          "payload"
        ]
      }`

**Comment by sinclairzx81** [comment]
@dsmyda Hi. Yes, there is a bit of flux going on currently with respect to `additionalProperties` handling in TypeBox, apologies. An update on `0.15.x` changed things to enforce `additionalProperties: false` for all object types. The intent was to mandate that all unexpected properties were caught, but there's been some downstream problems with this approach (both finding a reasonable schema representation as well as some typescript inference issues also).

In a future release (likely the next minor revision), I will be reverting things back such that  `additionalProperties` can be provided as a `CustomOption`. In the interim, I would recommend keeping on with `0.14.1` if you need configurable `additionalProperties`. 

Thanks for letting me know :)

**Comment by btisdall** [comment]
Hi and first of all thanks a lot for a really useful library! Regarding the issue at hand here, it seems the change to allow setting `additionalProperties` once again has not yet been made, are you able to give an update on when this will happen or should I just use 0.14.1 and stop worrying :) On a more general note, though this libary has such game-changing potential for node/TS development I slightly get the vibe that you haven't had much time to develop it recently and it would be super helpful if you could set expectations for users in this regard. Again, thanks a lot!

**Comment by sinclairzx81** [comment]
@btisdall Hey thanks. Yes, I do apologize for the delay on this. I have been putting off this update for sometime mainly due to it being a potential breaking change for some users, as well as being somewhat preoccupied with a few things inside and outside of work). But I am sorry for the delay. 

Just to let you know, I've just published a new version of TypeBox `0.17.0` which removes the `additionalProperties: false` constraint on `Type.Object({...})` and `Type.Intersect({ ... })` specifically. Users will now need to explicitly opt into `additionalProperties: false` to disallow additional properties appearing on the object. Documentation for this has been added to the project readme which you can read about here. https://github.com/sinclairzx81/typebox#AdditionalProperties

```typescript
const T = Type.Object({ a: Type.Number() }, { additionalProperties: false })
```
This update crosses over with a couple of other things going on related to  composition of `Type.Intersect({ ... })` and how TypeBox should express TypeScript intersections as schemas. I've added a bit of a write up to the PR for `0.17.0` that hopefully brings a little bit of insight into some of the reasoning going on there. You can read about it here https://github.com/sinclairzx81/typebox/pull/77.

> I slightly get the vibe that you haven't had much time to develop it recently and it would be super helpful if you could set expectations for users in this regard. Again, thanks a lot!

Would certainly be open to ideas on how to better communicate expectations to users. Would be curious to find out how other OSS projects manage this.

Many Thanks

**Comment by sinclairzx81** [comment]
Will close off this issue for the time being. @btisdall Feel free to respond to the above if you have suggestions on better communicating expectations. Thanks all.
